boot/makebootable.go

```diff
@@ -193,19 +187,49 @@ func makeBootable20(rootdir string, bootWith *BootableSet, bootFlags []string) e
 		return fmt.Errorf("internal error: cannot find bootloader: %v", err)
 	}
 
-	// record which recovery system is to be used on the bootloader, note
-	// that this goes on the main bootloader environment, and not on the
-	// recovery system bootloader environment, for example for grub
-	// bootloader, this env var is set on the ubuntu-seed root grubenv, and
-	// not on the recovery system grubenv in the systems/20200314/ subdir on
-	// ubuntu-seed
-	blVars["snapd_recovery_system"] = bootWith.RecoverySystemLabel
-	// always set the mode as install
-	blVars["snapd_recovery_mode"] = ModeInstall
+	if bootWith.Recovery {
+		// record which recovery system is to be used on the bootloader, note
+		// that this goes on the main bootloader environment, and not on the
+		// recovery system bootloader environment, for example for grub
+		// bootloader, this env var is set on the ubuntu-seed root grubenv, and
+		// not on the recovery system grubenv in the systems/20200314/ subdir on
+		// ubuntu-seed
+		blVars["snapd_recovery_system"] = bootWith.RecoverySystemLabel
+		blVars["snapd_recovery_mode"] = ModeInstall
+	} else {
+		blVars["snapd_recovery_mode"] = ModeRun
+	}
+
```

Comment:
this feels too implicit and unexpected, maybe blVars should also come from outside?